K2Bridge/KustoDAL/KustoDataAccess.cs

```diff
@@ -103,12 +107,38 @@ public async Task<IndexListResponseElement> GetIndexListAsync(string indexName)
                 using IDataReader kustoFunctions = await Kusto.ExecuteControlCommandAsync(readFunctionsCommand, RequestContext);
                 if (kustoFunctions != null)
                 {
-                    MapIndexList(kustoFunctions, response);
+                    while (kustoFunctions.Read())
+                    {
+                        tablesAndFunctions.Add(Convert.ToString(kustoFunctions[0]));
```

Comment:
what happens if there are no functions? always the error below will be thrown?

Reply:
Same as above, no element will be added, possibly ending up with an empty list and empty response. I could add some unit tests if you think that is necessary (I don't think these cases are covered by current tests).